cmake/modules/extensions.cmake

```diff
@@ -2196,6 +2200,24 @@ function(target_cc_option_fallback target scope option1 option2)
   endif()
 endfunction()
 
+function(target_ld_libraries target scope)
+  zephyr_get_parse_args(args ${ARGN})
+  list(REMOVE_ITEM ARGN NO_SPLIT)
+
+  foreach(libname ${ARGN})
+    if(args_NO_SPLIT)
+      set(libname ${ARGN})
+    endif()
+    string(JOIN "" option "-l" ${libname})
```

Comment:
This code part makes no sense now.

The NO_SPLIT is used in `target_ld_options()` to ensure that options are tested together in the `zephyr_check_compiler_flag()` call by setting `CMAKE_REQUIRED_FLAGS` to contain all options prior to that call.

As we are no longer calling `zephyr_check_compiler_flag()` then the logic of `NO_SPLIT` is not needed.

On top of that, CMake ensures that a call like `target_link_libraries(foo)` is translated in `-lfoo`*, meaning there is also need for the `JOIN .. "-l"` part.

This means this function can simply be written as:
```
function(target_ld_libraries target scope)
  target_link_libraries(${target} ${scope} ${ARGN})
endfunction()
```

Thus, the only reason for having `target_ld_libraries()` is to make a function similar to `target_ld_options()` but with a distinct name.

Also, it allows to extend the function later, if need be.

\* = **A plain library name:** The generated link line will ask the linker to search for the library (e.g. foo becomes -lfoo or foo.lib), ref: https://cmake.org/cmake/help/latest/command/target_link_libraries.html#overview.

Reply:
Yeah, the addition of `-l` serves to ensure that this function can only be used to add libraries and not other link options. If you don't think that's a useful developer aid, I'll remove it. Otherwise, yup, this function is getting pretty small :-)